library/std/src/os/uefi/ffi.rs

```diff
@@ -0,0 +1,73 @@
+use crate::ffi::{OsStr, OsString};
+use crate::sealed::Sealed;
+use crate::sys_common::ucs2;
+
+#[unstable(feature = "uefi_std", issue = "100499")]
+pub trait OsStrExt: Sealed {
+    /// This function does not do any allocation
+    fn to_ucs2<'a>(&'a self) -> ucs2::EncodeUcs2<'a>;
```

Comment:
This function is a hypothetically public trait method which returns an internal type so that would need to change too.

Reply:
Oh, I didn't know that `sys_common` was also private.